pkg/distributor/distributor_test.go

```diff
@@ -2228,43 +2225,70 @@ func TestDistributor_MetricsForLabelMatchers(t *testing.T) {
 	}
 
 	for testName, testData := range tests {
+		testData := testData
+
 		t.Run(testName, func(t *testing.T) {
-			now := model.Now()
+			t.Parallel()
 
-			// Create distributor
-			ds, ingesters, _, _ := prepare(t, prepConfig{
-				numIngesters:     numIngesters,
-				happyIngesters:   numIngesters,
-				numDistributors:  1,
-				shuffleShardSize: testData.shuffleShardSize,
-			})
+			for _, ingestStorageEnabled := range []bool{false, true} {
+				t.Run(fmt.Sprintf("ingest storage enabled: %t", ingestStorageEnabled), func(t *testing.T) {
+					now := model.Now()
```

Comment:
can you add t.Parallel() here as well so we run with and without ingest storage in parallel

Reply:
Done in https://github.com/grafana/mimir/pull/7393/commits/cff83c8f09893bfb14d253bb3c005be43444e554